File: exercise1/src/main/java/exercise1/EmptyFieldException.java
package exercise1;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(code = HttpStatus.NOT_FOUND)
public class EmptyFieldException extends RuntimeException {

	private static final long serialVersionUID = -5673170043066568594L;

	public EmptyFieldException() {
	}

	public EmptyFieldException(String message) {
		super(message);
	}

	public EmptyFieldException(Throwable cause) {
		super(cause);
	}

	public EmptyFieldException(String message, Throwable cause) {
		super(message, cause);
	}

}

File: New folder/Demo - Kotlin/src/main/kotlin/demo/EmptyFieldException.kt
package demo

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus


@ResponseStatus(code = HttpStatus.NOT_FOUND)
class EmptyFieldException : RuntimeException {
    constructor() {}
    constructor(message: String?) : super(message) {}
    constructor(cause: Throwable?) : super(cause) {}
    constructor(message: String?, cause: Throwable?) : super(message, cause) {}

    companion object {
        private const val serialVersionUID = -5673170043066568594L
    }
}
